src/Sylius/Bundle/AdminBundle/Command/AbstractAdminUserCommand.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+abstract class AbstractAdminUserCommand extends Command
```

Comment:
Inheritance shouldn't be used just for code/dependency reuse.

https://en.wikipedia.org/wiki/Composition_over_inheritance

Sorry, should've seen it at previous review.

---
Each command should define each own `SymfonyStyle $io` with `initialize`.

`createEmailQuestion` and `createQuestionWithNonBlankValidator` can be moved either to `QuestionTrait` or to `QuestionFactory` as both methods are creational and not particularly locked to `AdminUser`. 

```php
namespace Sylius\Bundle\AdminBundle\Command\Factory;

use Symfony\Component\Console\Question\Question;

class QuestionFactory implements QuestionFactoryInterface
{
    public function createEmailQuestion(): Question;
    public function createQuestionWithNonBlankValidator(string $askedQuestion, bool $hidden = false): Question;
}
```

WDYT?

Reply:
P.S. I've update creational method names to more sylius common factory naming and plus `createQuestionWithNonBlankValidator` doesn't check for `NonBlank` value, but only `NotNull`

https://symfony.com/doc/current/reference/constraints/NotBlank.html

Reply:
Removed the 2nd blank line and renamed the tests.

For the other changes, I don't really get the point to be honest. The methods I moved to the abstract class have been implemented in #14665 and have not been changed in this PR. I just thought it made sense to reuse them, and thus, I introduced a parent class. I don't see why relying on a trait would improve this particular code at all. But if you feel this is a must to get the PR merged, I'll happily do it.